annoyed-alligators/socl_media/templates/terminal.html

```diff
@@ -0,0 +1,112 @@
+  <script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>
+  <script src="https://unpkg.com/axios/dist/axios.min.js"></script>
+  {% comment %} <script type='text/javascript'>
+    var people = {{ items|safe }};
+  </script> {% endcomment %}
+
+  <div id="app">
+    {% comment %} <input type="text" v-model="command">
+    <button @click="run_command">Submit</button>
+    <br>
+    
+    <br> {% endcomment %}
+
+    <div id="terminal">
+      <div id="terminal-window">
+        <div id="terminal-output" v-html="response">
+        </div>
+      </div>
+      <div id="terminal-input"> 
+        ~ $ <input type="text" 
+              class="terminal-input" 
+              autofocus="autofocus" 
+              @keyup.enter="run_command" 
+              @keyup.38="populate_previous"
+              @keyup.40="clear_previous"
+              v-model="command">
+      </div>
+    </div>
+
+  </div>
+
+  
+
+  <script>
+    var app = new Vue({
+      delimiters: ['[[', ']]'],
+      el: '#app',
+      data: {
+        command: '',
+        previous_command_array: [],
+        current_command: 0,
+        response: '',
+      },
+
+      methods: {
+        populate_previous: function(){
+          if(this.current_command < this.previous_command_array.length){
+            this.command = this.previous_command_array[this.current_command];
+            this.current_command += 1;
+          }
+        },
+        clear_previous: function(){
+          if(this.current_command > 0){
+            this.current_command -= 1;
+            this.command = this.previous_command_array[this.current_command-1];
+          }
+        },
+        run_command: function(){
+          axios.get(`/terminal/command?c=${this.command}`)
+          .then(res =>{
+            this.response = this.response + res.data + "<br>"
+          }, (error) =>{
+            console.log(error)
+          }) 
+          this.previous_command = this.command;
+          this.previous_command_array.unshift(this.command)
+          this.command = '';
+          console.log(this.previous_command_array)
+        }
+      }
+    });
+  </script>
+
+  <style>
```

Comment:
Internal CSS is not so preferred. We'll have a separate CSS file for all the styles.
Make a folder called `css` inside the `static` folder, and add this CSS to `static/css/styles.css`.

Reply:
Vue.js, components have all three parts bundled in a single page, i.e., HTML, script, and CSS. It's known as a single file component. You can read about it here: https://vuejs.org/v2/guide/single-file-components.html
I'd prefer to keep it the Vue.js components laid out this way. I'll add the scoped attribute to the style tags.

Reply:
Oh okay, I didn't know that. It's fine then!